### PR TITLE
[doc] Add a table to map release tags from the compiler to SwiftSyntax.

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please test

File: README.md [hunk 1]
```diff
@@ -26,7 +26,13 @@ let package = Package(
 )
 ```
 
-Replace `<#Specify Release tag#>` by the version of SwiftSyntax that you want to use. Tags will be created for every release of the compiler in the form `swift-4.2-RELEASE` and for every nightly build in the form `swift-4.2-DEVELOPMENT-SNAPSHOT-2018-08-25-a`.
+Replace `<#Specify Release tag#>` by the version of SwiftSyntax that you want to use (see the following table for mapping details). Tags will be created for every release of the compiler in the form `swift-4.2-RELEASE` and for every nightly build in the form `swift-4.2-DEVELOPMENT-SNAPSHOT-<DATE>`.
```
Comment (akyrtzi) [inline README.md]: `Tags will be created for every...` is confusing and does not reflect the change in the tagging scheme, recommend to change to:
```
Tags will be created for every nightly build in the form `<SwiftSyntax Release Tag>-DEVELOPMENT-SNAPSHOT-<DATE>`.
```

**Comment by nkcsgexi** [comment]
@akyrtzi Changed. Mind taking another look?

**Comment by mdiep** [comment]
I'm a little confused personally. Does this need to match the version of `swift` that you're using?

**Comment by nkcsgexi** [comment]
@mdiep That's correct. Swift-syntax relies on compiler (swiftc) to generate syntax trees to deserialize.  For every major releases of Swift, Syntax tree generated from the compiler can vary structurally. So we have to agree on compatible versions so the parser can generate the tree swift-syntax recognizes, and vice versa. If you have any trouble building your tool on top of swift-syntax, please let us know :)

**Pull request merged**